src/clj_commons/byte_streams/graph.clj

```diff
@@ -31,8 +31,8 @@
       (= type (.type ^Type x))))
   (hashCode [_]
     (bit-xor
-      (hash wrapper)
-      (hash type)))
+      ^long (hash wrapper)
+      ^long (hash type)))
```

Comment:
Hmmm. `hash` returns an int, not a long. But the compiler uses the hint to force a cast. Not sure which is faster, though.

Before:
```java
    public int hashCode() {
        return RT.intCast(Numbers.xor(((IFn)const__4.getRawRoot()).invoke(this.wrapper), ((IFn)const__4.getRawRoot()).invoke(this.type)));
    }
```

After:
```java
    public int hashCode() {
        return RT.intCast(RT.uncheckedLongCast((Number)((IFn)const__4.getRawRoot()).invoke(this.wrapper)) ^ RT.uncheckedLongCast((Number)((IFn)const__4.getRawRoot()).invoke(this.type)));
    }
```

Truthfully, the best thing is to skip hints here and use explicit conversions to match the types involved. Try:

```clojure
(hashCode [_]
    (int
      (bit-xor
        (long (hash wrapper))
        (long (hash type)))))
```

Reply:
Does Clojure even allow ^int type hints? It gives me an "Only long and double primitives are supported" error.

I'll change to explicit cast.

Reply:
`^int` is a valid type hint, but not for function parameters or return values. It's fine in local contexts, like a `let`.

Hell, it used to be (iirc) that unboxed parameters weren't allowed at all, but it seems like they're allowed for fn arities up to 4 now.

Reply:
(You shouldn't need to hint `hashCode` btw, that's defined by `Object`.)